File: 0678/678.go
func validParenthesisString(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := validParenthesisString(root.Left)
	r := validParenthesisString(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0678/678.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var validParenthesisString = function(root) {
    if (!root) return 0;
    return 1 + Math.max(validParenthesisString(root.left), validParenthesisString(root.right));
};